cpp/src/io/orc/reader_impl.cu

```diff
@@ -989,12 +68,22 @@ table_with_metadata reader::impl::read(uint64_t skip_rows,
   out_metadata.user_data = {out_metadata.per_file_user_data[0].begin(),
                             out_metadata.per_file_user_data[0].end()};
 
-  // Select only stripes required (aka row groups)
-  auto const [rows_to_skip, rows_to_read, selected_stripes] =
-    _metadata.select_stripes(stripes, skip_rows, num_rows_opt, _stream);
+  // Save the output table metadata into `_output_metadata` for reuse next time.
+  _output_metadata = std::make_unique<table_metadata>(out_metadata);
+
+  return out_metadata;
+}
+
+table_with_metadata reader::impl::read_chunk_internal()
+{
+  // There are no columns in the table
+  if (_selected_columns.num_levels() == 0) { return {std::make_unique<table>(), table_metadata{}}; }
```

Comment:
Another one mostly tangential to the PR: we should probably still return user data in the table_metadata, even if no columns are selected/exist.
Edit: should be an easy fix, we just need to call `populate_metadata` before this line and return `std::move(out_metadata)`.

Reply:
Humnn, this seems to be wrong:
```
FAILED python/cudf/cudf/tests/test_orc.py::test_orc_columns_and_index_param[columns1-True-None] - IndexError: list index out of range
FAILED python/cudf/cudf/tests/test_orc.py::test_orc_columns_and_index_param[columns1-True-index_obj1] - IndexError: list index out of range
FAILED python/cudf/cudf/tests/test_orc.py::test_orc_columns_and_index_param[columns1-True-index_obj2] - IndexError: list index out of range
FAILED python/cudf/cudf/tests/test_orc.py::test_orc_columns_and_index_param[columns1-None-None] - AssertionError: DataFrame are different

DataFrame shape mismatch
[left]:  (0, 0)
[right]: (3, 0)
FAILED python/cudf/cudf/tests/test_orc.py::test_orc_columns_and_index_param[columns1-None-index_obj1] - IndexError: list index out of range
FAILED python/cudf/cudf/tests/test_orc.py::test_orc_columns_and_index_param[columns1-None-index_obj2] - IndexError: list index out of range
```

Reply:
Looks like Python uses something from the the table_metadata that isn't correct for this case.
Let's leave this one for another PR :)